### PR TITLE
feat : update api nodes to use jwtToken innerAPI

### LINKED ISSUE #9: feat : add verifying process when load comments or articles
add check whether user is belonging to target community.

feat : add API node which return community id from comment id

### PR DESCRIPTION
**Pull request by ghost**
@Malang2  new JWT Verification Process

Closes #9

### CODE REVIEW

File: routes/jwtauth/jwtgetUser.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            result : _response,
+        });
+        await createLog.save(async (err) => {
+            if (err) console.error(err);
+        });
+        return _response;
+    };
+
+    //#HANDLE THIS NODE MORE WHEN OPTIONS ARE EXIST
+    if (!(objectID === undefined || objectID === null)) {
```
Comment (YunYeongChang) [inline routes/jwtauth/jwtgetUser.js:47]: Codacy found an issue: ['objectID' is not defined.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

**Comment by YunYeongChang** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- routes/jwtauth/jwtgetUser.js  3
         

Clones added
============
- routes/post/new_article.js  1
- routes/post/mod_article.js  1
- routes/jwtauth/jwtRefresh.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533&bid=20289696)

**Comment by YunYeongChang** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- routes/jwtauth/jwtgetUser.js  3
         

Clones added
============
- routes/post/mod_article.js  1
- routes/jwtauth/jwtRefresh.js  1
- routes/post/new_article.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533&bid=20289696)

**Review by rycont — DISMISSED** [review]
LGTM✌

File: models/jwtblock.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { model, Schema } from "mongoose";
+
+const jwtBlock = new Schema({
+  "blockdate": String,
```
Comment (rycont) [inline models/jwtblock.js:4]: Doublequotes not required!

Comment (YunYeongChang) [reply models/jwtblock.js:4]: @rycont Thank you for helpful information

**Review by MinSeungHyun — DISMISSED** [review]

**Comment by Malang2** [comment]
great

File: routes/jwtauth/jwtRefresh.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import { verify as _verify } from "jsonwebtoken";
+import moment from "moment";
+import { getClientIp } from "request-ip";
+import jwtVerify from "./jwtVerify";
+import jwtBlock from "./jwtBlock";
+import { getNewSignedAccessJWTToken as jwtSign } from "./jwtSign";
+import authLog from "../../models/authlog";
+import User from "../../models/user";
+  
+const jwtRefresh = async (req, accesstoken, refreshtoken) => {
```
Comment (YunYeongChang) [inline routes/jwtauth/jwtRefresh.js:10]: Codacy found an issue: [Async arrow function has a complexity of 8. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

**Comment by YunYeongChang** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- routes/post/load_comments.js  7
- routes/post/new_article.js  11
- routes/post/mod_article.js  12
- routes/post/load_articles.js  7
- routes/jwtauth/jwtVerifyAccess.js  17
- routes/post/load_editlogs.js  10
- routes/jwtauth/jwtgetUser.js  9
- routes/post/mod_comment.js  9
- routes/post/new_comment.js  8
- routes/jwtauth/jwtVerify.js  7
         

Clones added
============
- routes/post/load_comments.js  1
- routes/post/new_article.js  1
- routes/post/mod_article.js  2
- routes/post/load_articles.js  1
- routes/jwtauth/jwtRefresh.js  1
- routes/post/new_comment.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533&bid=20289696)

File: routes/jwtauth/jwtVerifyAccess.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+var _response = {"privileges":null, "verify_error": "ERR_VERIFY_USER_SYNTAX_FAILED"};
+
+const returnResult = (result, error) => {
+    if (result !== undefined) _response.privileges = result;
+    if (error !== undefined) _response.verify_error = error;
+    return _response;
+};
+
+const jwtVerifyAccess = async (userid, type, target) => {
```
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerifyAccess.js:14]: Codacy found an issue: [Async arrow function has a complexity of 17. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerifyAccess.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            if (_comment === undefined || _comment === null) return returnResult(null, "ERR_JWT_CAN_NOT_FIND_COMMENT");
+            if (!_comment) return returnResult(null, _comment);
+            target = _comment.target.community;
+        }
+        const _community = await Community.findOne({"_id": target, "status":"inservice"}, {"info":0});
+        if (_community === null || _community === undefined) return returnResult(null, "ERR_JWT_CAN_NOT_FIND_COMMUNITY");
+        if (!_community) return returnResult(null, _community);
+
+        //#FIND USER PRIVILEGE IN COMMUNITY MEMBER LIST
+        const index = await _community.userlist.findIndex(obj => obj.user === userid);
```
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerifyAccess.js:33]: Codacy found an issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerifyAccess.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        if (index === -1) return returnResult(null, "ERR_JWT_NOT_USER_IN_COMMUNITY");
+
+        const _privilege = await Privilege.findOne({"_id":_community.userlist[index].privilege, "user":userid});
```
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerifyAccess.js:36]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerify.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { verify as _verify } from "jsonwebtoken";
+import jwtBlock from "../../models/jwtblock";
+import User from "../../models/user";
+  
+const jwtVerify = async (token, ignoreblock) => {
```
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerify.js:5]: Codacy found an issue: [Async arrow function has a complexity of 7. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerifyAccess.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        if (!privilege) return returnResult(null, _privilege);
+        return returnResult(privilege.permission, null);
```
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerifyAccess.js:38]: Codacy found an issue: ['privilege' is not defined.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerifyAccess.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        const _community = await Community.findOne({"_id": target, "status":"inservice"}, {"info":0});
+        if (_community === null || _community === undefined) return returnResult(null, "ERR_JWT_CAN_NOT_FIND_COMMUNITY");
+        if (!_community) return returnResult(null, _community);
+
+        //#FIND USER PRIVILEGE IN COMMUNITY MEMBER LIST
+        const index = await _community.userlist.findIndex(obj => obj.user === userid);
+        if (index === -1) return returnResult(null, "ERR_JWT_NOT_USER_IN_COMMUNITY");
+
+        const _privilege = await Privilege.findOne({"_id":_community.userlist[index].privilege, "user":userid});
+        if (!privilege) return returnResult(null, _privilege);
```
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerifyAccess.js:37]: Codacy found an issue: ['privilege' is not defined.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtgetUser.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
+import jwtVerify from "./jwtVerify";
+import { getClientIp } from "request-ip";
+import User from "../../models/user";
+import authLog from "../../models/authlog";
+import jwtVerifyAccess from "./jwtVerifyAccess";
+import moment from "moment";
+
+const jwtgetUser = async (req, jwtToken, options) => {
```
Comment (YunYeongChang) [inline routes/jwtauth/jwtgetUser.js:8]: Codacy found an issue: [Async arrow function has a complexity of 9. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

**Comment by YunYeongChang** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 9
           

Complexity increasing per file
==============================
- routes/post/load_comments.js  9
- routes/post/load_editlogs.js  12
- routes/jwtauth/jwtgetUser.js  9
- routes/post/mod_comment.js  11
- routes/post/new_article.js  13
- routes/jwtauth/jwtRefresh.js  11
- routes/post/mod_article.js  14
- routes/jwtauth/jwtVerify.js  7
- routes/jwtauth/jwtVerifyAccess.js  17
- routes/post/load_articles.js  9
- routes/post/new_comment.js  10
         

Clones added
============
- routes/post/load_comments.js  3
- routes/post/load_editlogs.js  2
- routes/post/mod_comment.js  1
- routes/post/new_article.js  2
- routes/jwtauth/jwtRefresh.js  1
- routes/post/mod_article.js  3
- routes/post/load_articles.js  3
- routes/post/new_comment.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533&bid=20289696)

File: routes/jwtauth/jwtRefresh.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
```
Comment (YunYeongChang) [inline routes/jwtauth/jwtRefresh.js:10]: Codacy found an issue: [Async arrow function has a complexity of 8. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerifyAccess.js (hunk 8 above)
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerifyAccess.js:38]: Codacy found an issue: ['privilege' is not defined.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerify.js (hunk 7 above)
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerify.js:5]: Codacy found an issue: [Async arrow function has a complexity of 7. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerifyAccess.js (hunk 6 above)
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerifyAccess.js:36]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerifyAccess.js (hunk 5 above)
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerifyAccess.js:33]: Codacy found an issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/post/load_comments.js [hunk 12]
```diff
@@ -1,23 +1,32 @@
...
+import jwtgetUser from "../jwtauth/jwtgetUser";
+import jwtRefresh from "../jwtauth/jwtRefresh";
 import { db_error } from "../../app";
 import mongoose from "mongoose";
 import responseFunction from "../coms/apiResponse";
 import Comment from "../../models/post/comment";
 
 const router = Router();
-router.get ("/:target", async (req,res) => {//#CHECK DATABASE STATE AND WHETHER PROVIDED POST DATA IS VALID 
+router.get ("/:target", async (req,res) => {
```
Comment (YunYeongChang) [inline routes/post/load_comments.js:10]: Codacy found an issue: [Async arrow function has a complexity of 7. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtgetUser.js (hunk 10 above)
Comment (YunYeongChang) [inline routes/jwtauth/jwtgetUser.js:8]: Codacy found an issue: [Async arrow function has a complexity of 9. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerifyAccess.js (hunk 9 above)
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerifyAccess.js:37]: Codacy found an issue: ['privilege' is not defined.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerifyAccess.js (hunk 4 above)
Comment (YunYeongChang) [inline routes/jwtauth/jwtVerifyAccess.js:14]: Codacy found an issue: [Async arrow function has a complexity of 17. Maximum allowed is 4.](https://app.codacy.com/gh/implude/local-community-backend/pullRequest?prid=6225533)

File: routes/jwtauth/jwtVerifyAccess.js (hunk 6 above)
Comment (YunYeongChang) [reply routes/jwtauth/jwtVerifyAccess.js:36]: solved

File: routes/jwtauth/jwtVerifyAccess.js (hunk 5 above)
Comment (YunYeongChang) [reply routes/jwtauth/jwtVerifyAccess.js:33]: solved

File: routes/jwtauth/jwtVerifyAccess.js (hunk 9 above)
Comment (YunYeongChang) [reply routes/jwtauth/jwtVerifyAccess.js:37]: solved

**Review by MinSeungHyun — APPROVED** [review]
👍

**Review by alsh02 — APPROVED** [review]
so beautiful code

**Pull request merged**